test/jdk/java/lang/ProcessBuilder/PipelineLeaksFD.java

```diff
@@ -68,11 +68,18 @@ void checkForLeaks(List<ProcessBuilder> builders) throws IOException {
             Assert.fail("There should be at least 3 pipes before, (0, 1, 2)");
         }
 
+        // Redirect all of the error streams to stdout (except the last)
+        // so those file descriptors are not left open
+        for (int i = 0; i < builders.size() - 1; i++) {
+            builders.get(i).redirectErrorStream(true);
+        }
```

Comment:
This loop could be removed by doing redirects in the data provider.

Reply:
Thanks for the review.
The effect would be the same, though it would distribute the needed constraint to more places in the source and the descriptive comment would need to be duplicated.